src/core/score_map.h

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <optional>
+#include <string_view>
+
+#include "core/dense_set.h"
+
+extern "C" {
+#include "redis/sds.h"
+}
+
+namespace dfly {
+
+namespace detail {
+
+class SdsScorePair {
+ public:
+  SdsScorePair(sds k, double v) : first(k), second(v) {
+  }
+
+  SdsScorePair* operator->() {
+    return this;
+  }
+
+  const SdsScorePair* operator->() const {
+    return this;
+  }
+
+  const sds first;
+  const double second;
+};
```

Comment:
Whats the point of this class 🤔

Reply:
to provide iterator UX.
you are right that template are better in this case (limited set of predefined types).
I do not have time for this but if you want to do it - feel free to improve.